server.go

```diff
@@ -30,8 +32,23 @@ type ServerOptions struct {
 	Authorization     string
 	Placeholder       string
 	PlaceholderImage  []byte
+	Endpoints         Endpoints
 	AlloweOrigins     []*url.URL
-	MaxAllowedSize    int
+}
+
+// Endpoints represents a list of endpoint names to disable.
+type Endpoints []string
+
+// IsValid validates if a given HTTP request endpoint is valid or not.
+func (e Endpoints) IsValid(r *http.Request) bool {
+	parts := strings.Split(r.URL.Path, "/")
+	endpoint := parts[len(parts)-1]
+	for _, name := range e {
+		if endpoint == name {
+			return false
+		}
+	}
+	return true
```

Comment:
I actually think this approach is beautiful and good if we want to have a better control on every requests.

But if we want a case where only a small amount of request will need to be disabled 
 then we have some overhead for requests which are allow by default, but now it has to run this code.

Should we switch to another approach? Where we chose if handler will need to check if can be disabled? So this kind of check will not get run on normal GET request or resize request, only in special request such as POST, form upload we will run this check?

Reply:
I see your point. The overhead is relatively speaking insignificant. I have considered an alternative implementation which is not based on a "middleware" http.Handler that won't introduce any overhead, but I prefered this approach since it's more scalable, simple and easy to maintain. I also discard using memoization strategies given the tiny overhead.

> Should we switch to another approach? Where we chose if handler will need to check if can be disabled? So this kind of check will not get run on normal GET request or resize request, only in special request such as POST, form upload we will run this check?

I don't think we need more complexity here. At the end, the goal is to easily enable/disable server capabilities exposed as HTTP API. More complex scenarios would need to be implemented separately. I'm also considering supporting imaginary plugins as Go pre-compiled shared libraries that would enable developers programmatically augment imaginary without pre-compiling it. See: #159